sdk/storage/Azure.Storage.DataMovement/tests/CleanUpTransferTests.cs

```diff
@@ -124,7 +124,7 @@ public async Task CleanupAfterFailureAsync()
                 It.IsAny<CancellationToken>()),
                 Times.Once());
             destMock.Verify(b => b.DeleteIfExistsAsync(It.IsAny<CancellationToken>()),
-                Times.Once());
+                Times.Never());
```

Comment:
This test and the one below seemed to be asserting incorrectly for calling `DeleteIfExistsAsync` based on the comment [here](https://github.com/Azure/azure-sdk-for-net/blob/ef9e6bb7d125031fbb1a7ed25521eaa4143b9f48/sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs#L453-L455). Can you confirm @amnguye ?

Reply:
Let me look into this. I would have expected for this not to change based on the change you made.

Because the test doesn't have overwrite enabled, `DeleteIfExistsAsync` should be still called here since the error was `AuthenticationFailed` in this specific test.

Reply:
The reason I think that it may have been wrong before is that according to [this logic](https://github.com/Azure/azure-sdk-for-net/blob/ef9e6bb7d125031fbb1a7ed25521eaa4143b9f48/sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs#L599), we should only call `DeleteIfExistsAsync` when the job failure type is `JobPartFailureType.Other`.

Reply:
Yes you're right. We shouldn't be calling `DeleteIfExistsAsync` in this case. Especially since it's a authentication failure, calling `DeleteIfExistsAsync` would also be pointless here since we can't authenticate anyways to call delete.

Thanks for correcting this.